engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1375,28 +1434,47 @@ private void processOperatorDeployStatus(final PTOperator oper, OperatorHeartbea
         break;
       case PENDING_UNDEPLOY:
         if (ds == null) {
+          long groupId = 0;
+          ContainerRedeployRequest request = getStopOperatorGroupId(oper, ohb);
```

Comment:
The logic of these 5 lines can be combined into a single function call (say redeployOperatorInGroup). What this frunction does is basically find the request using operatorsToUndeploy set and in that request add the same operator to operatorsToDeploySet and then returns the groupId or it could also generate the StopOperatorEvent in the function. A comment can explain what the function does